app/src/main/java/Fragments/ProfileFragment.java

```diff
@@ -129,24 +152,47 @@ private void editPhoto() {
             @Override
             public void onComplete(@NonNull Task<Void> task) {
                 if (task.isSuccessful()) {
-                    Log.d("update_user", "User profile photo updated.");
+                    Log.d("USER_UPDATE", "User profile photo updated.");
                 }
             }
         });
     }
 
-    private void editName() {
-        UserProfileChangeRequest updateName = new UserProfileChangeRequest.Builder()
-                .setDisplayName("Evan Mateo") // TODO: Implement an "Edit Name" fragment
-                .build();
-
-        user.updateProfile(updateName).addOnCompleteListener(new OnCompleteListener<Void>() {
-            @Override
-            public void onComplete(@NonNull Task<Void> task) {
-                if (task.isSuccessful()) {
-                    Log.d("update_user", "User profile name updated.");
+    private void updateName() {
+        if (name_tv.getText().toString().trim().equals("")) {
+            Toast.makeText(view.getContext(), "This is not a valid name", Toast.LENGTH_SHORT).show();
+        } else {
+            name = name_tv.getText().toString();
+            UserProfileChangeRequest updateName = new UserProfileChangeRequest.Builder()
+                    .setDisplayName(name)
+                    .build();
+
+            user.updateProfile(updateName).addOnCompleteListener(new OnCompleteListener<Void>() {
+                @Override
+                public void onComplete(@NonNull Task<Void> task) {
+                    if (task.isSuccessful()) {
+                        Log.d("USER_UPDATE", "User profile name updated.");
+                    }
                 }
-            }
-        });
+            });
+        }
+
+    }
+
+    private void updateEmail() {
+        if (name_tv.getText().toString().trim().equals("")) {
+            Toast.makeText(view.getContext(), "This is not a valid name", Toast.LENGTH_SHORT).show();
+        } else {
```

Comment:
updateEmail() does not work because you're getting the name_tv instead of email_tv.

Reply:
Yes. The username is the email. I've tested it and when I successfully change my email, firebase sends an email notification with an option to revert the change.

Reply:
Good catch Evan. I just noticed something with Firebase's updateEmail though: seems like it doesn't update if it's not Gmail that's being used, not sure if that's intended. I was trying with Hotmail and it wasn't working, gotta look more into that, or we could just stick with Gmail?

Reply:
I tested that and it should work with any email, so long as it's formatted correctly. i.e. something@###.###